rdkit/Chem/EnumerateStereoisomers.py

```diff
@@ -66,30 +66,36 @@ def __init__(self, group):
     self._original_parities = [(a, a.GetChiralTag()) for a in group.GetAtoms()]
 
   def flip(self, flag):
-    if flag:
-      for a, original_parity in self._original_parities:
-        a.SetChiralTag(original_parity)
-    else:
-      for a, original_parity in self._original_parities:
-        if original_parity == Chem.ChiralType.CHI_TETRAHEDRAL_CW:
-          a.SetChiralTag(Chem.ChiralType.CHI_TETRAHEDRAL_CCW)
-        elif original_parity == Chem.ChiralType.CHI_TETRAHEDRAL_CCW:
-          a.SetChiralTag(Chem.ChiralType.CHI_TETRAHEDRAL_CW)
-
+    CHIRAL_DICT = {Chem.ChiralType.CHI_TETRAHEDRAL_CW: Chem.ChiralType.CHI_TETRAHEDRAL_CCW,
+                   Chem.ChiralType.CHI_TETRAHEDRAL_CCW: Chem.ChiralType.CHI_TETRAHEDRAL_CW, 
+                   # Chem.ChiralType.CHI_UNSPECIFIED: Chem.ChiralType.CHI_UNSPECIFIED,
+                   # Chem.ChiralType.CHI_OTHER: Chem.ChiralType.CHI_OTHER,
+                   }
+    for atom, original_parity in self._original_parities:
+      try:
+        if flag:
+          chirality = original_parity 
+        else:
+          chirality = CHIRAL_DICT[original_parity]
+        atom.SetChiralTag(chirality)
+      except KeyError:
+        pass
+    del CHIRAL_DICT
```

Comment:
Deleting the dictionary shouldn't be necessary. However, I think the modified code is less efficient. 
- Constructing and deleting the dictionary costs time.
- If flag is True, the `for` loop has no additional condition (likely not critical)
- If flag is False, `atom.setChiralTag` is only called for atoms that are chiral. In the new code, it is called for all atoms.

Reply:
Agree, I would reverse it back to originial